### DISCUSSION TITLE
Excessive (3X) ram usage during parquet generation

### DISCUSSION THREAD

**Discussion by WyattLiu** [duckdb/duckdb#9176] [General]
```
<schema>
COPY LINEITEM FROM '/mnt/dbgen/150G_TPCH/lineitem.tbl' (DELIMITER '|');
COPY (SELECT * FROM LINEITEM) TO '/mnt/dbgen/150G_TPCH/lineitem.parquet' (FORMAT 'parquet')
```

I am trying to cook some parquet from a reasonably large TPCH, and it turns out I used 331.8745G of RAM.
The single table lineitem 110.84G in tbl/csv, I just don't get why we need to triple the ram. I am on dev 0.8.1.
Could this be the default one that comes with some compression?

I discovered the default max ram is 101G ish which can be overridden by setting a higher limit, which I hope should be a higher value.

Regards.

**Comment by WyattLiu** [comment]
I am trying 
```
<schema>
COPY LINEITEM FROM '/mnt/dbgen/150G_TPCH/lineitem.tbl' (DELIMITER '|');
COPY LINEITEM TO '/mnt/dbgen/150G_TPCH/lineitem.parquet' (FORMAT 'parquet')
```

**Reply by Tishj** [reply]
Try doing it streaming instead, now you're creating a table and then copying that

This might work:
```sql
COPY (SELECT * FROM READ_CSV('/mnt/dbgen/150G_TPCH/lineitem.tbl', delimiter='|')) TO '/mnt/dbgen/150G_TPCH/lineitem.parquet' (FORMAT 'parquet');
```

**Reply by WyattLiu** [reply]
I only do the COPY and schema because I want to bake the variable types (int32 vs int64 etc) into the parquet file. The current duckdb extension does a great job after I checked with parquet-tools inspect.

Do you know how to stream with schema defined?

After I removed the SELECT* would save 1x. But still @ 2X without streaming, streaming should work.

**Reply by Tishj** [reply]
The `columns` parameter should help with this
<https://duckdb.org/docs/data/csv/overview.html>
